### DISCUSSION TITLE
GTK3 popupinfo window color consistency and systray consistency

### DISCUSSION THREAD

**Discussion by dmillfree868** [audacious-media-player/audacious#92] [General]
I noticed If I compile audacious with gtk3 that the popupinfo window is light gray with black text. Shouldn't this be more consistent with the same look and colors as the gtk2 version or the qt version? The qt version the size of the popupinfo window is too big and the systray icon shows no popupinfo windows at all. I know there is ongoing development so it's understandable.

**Comment by jlindgren90** [comment]
GTK3 broke a lot of custom theming, that's one of the reasons we avoided it for a long time.

**Comment by radioactiveman** [comment]
Regarding the missing popup info for the Qt status icon: The code to handle this is there but the [event method](https://github.com/audacious-media-player/audacious-plugins/blob/a5d02335b02fdf9cfc6815a33e7be3812e0d6f24/src/statusicon-qt/statusicon.cc#L158-L174) seems to be never called.
Does this work on your machine @jlindgren90? Maybe it's related to the desktop environment.

**Comment by dmillfree868** [comment]
I was thinking the same thing. I have audacious installed on Slackware
Linux using the xfce4 desktop. Slackware offers the KDE desktop which is Qt
based but I don't install that or use it.

**Comment by jlindgren90** [comment]
Qt is probably using StatusNotifierItem rather than XEmbed for the status icon. SNI doesn't support all the same event types as XEmbed.

**Comment by Npa972** [comment]
Changing the volume with the mouse also don't work with qt version.
Actually i use a gtk3 git version and i can't see the colors problem in the popup windows (playlist or systray icon). Perhaps because i use a dark theme ?
Changing the volume on systray works fine with gtk3.
And the new hamburger menu is nice with symbolic icons.
System : archlinux with xfce, papirus-icon-theme and adwaita-dark
Another thing is gtk3 version consume much less memory than qt version.
Thanks to the developpers

**Comment by Npa972** [comment]
![image](https://github.com/audacious-media-player/audacious/assets/49346431/46d58c25-6ab8-4e4b-801c-c4cece748e2a)

**Comment by Npa972** [comment]
@dmillfree868, ok i understand now what you mean :
audacious-git gtk2 : black background with gradient and with white text
![image](https://github.com/audacious-media-player/audacious/assets/49346431/a86782e4-e8a3-4b69-b4b0-004af942d4ce)
audacious-git gtk3 : light gray background without gradient and with black text. It's less visible with a dark theme
![image](https://github.com/audacious-media-player/audacious/assets/49346431/413b5121-f799-4453-9738-d27e7a67e027)

**Reply by dmillfree868** [reply]
Sorry I should have posted a screenshot. I actually took the time to install KDE temporarily (Yikes what a mess) and that made no difference with qt modes tray icon. I just reverted back to building audacious with gtk2. Problem solved.

**Comment by radioactiveman** [comment]
The GTK3 info popup now looks the same as with GTK2 / Qt, implemented with https://github.com/audacious-media-player/audacious/commit/fcdb46ddacaf779d91fda605e51472e084b67dbe.

![audacious-gtk3](https://github.com/audacious-media-player/audacious/assets/1301412/43a8a2b6-8bca-4421-b104-1dfe3c2b6dae)

**Comment by dmillfree868** [comment]
![12-04-2023_09:07am](https://github.com/audacious-media-player/audacious/assets/95503349/de97f585-ee3a-4ebc-b298-b153efdda6f2)
Wow you guys are awesome, your work is really appreciated.

**Comment by Npa972** [comment]
I built the latest version with gtk3 and it's perfect.
Thank's
